javers-core/src/test/groovy/org/javers/core/JaversRepositoryShadowE2ETest.groovy

```diff
@@ -669,6 +664,52 @@ class JaversRepositoryShadowE2ETest extends JaversRepositoryE2ETest {
         assertThinShadowOfPhone(shadows.first().shallowPhonesMap["key"])
     }
 
+    @Unroll
+    def "should query for #what shadows by multiple GlobalIds with limit"() {
+        given:
+        objects.each {
+            javers.commit("author", it)
+        }
+
+        when:
+        def shadows = javers.findShadows(query)
+
+        then:
+        shadows.size() == 6
+        shadows.each {
+            def value = it.it
+            assert expectedId.contains(value)
+        }
+
+        where:
+        what << ["Entity", "Unbounded ValueObject", "Bounded ValueObject"]
+        objects << [
+                (1..5).collect { new SnapshotEntity(id: 1, intProperty: it) }
+                        + (1..5).collect { new SnapshotEntity(id: 2, intProperty: it) }
+                        + new SnapshotEntity(id: 3), //noise
+                (1..5).collect { new DummyAddress(city: "London${it}") }
+                        + (1..5).collect { new DummyPoint(it, 2) }
+                        + new DummyPoint(6, 2), //noise
+                (1..5).collect { new SnapshotEntity(id: 1, valueObjectRef: new DummyAddress(city: "London${it}")) }
+                        + (1..5).collect { new SnapshotEntity(id: 2, valueObjectRef: new DummyAddress(city: "Paris${it}")) }
+                        + new SnapshotEntity(id: 3, valueObjectRef: new DummyAddress(city: "London1")) //noise
+        ]
+        query << [
+                [byInstanceId(1, SnapshotEntity).limit(3).build(), byInstanceId(2, SnapshotEntity).limit(3).build()],
+                [QueryBuilder.byClass(DummyAddress).limit(3).build(), QueryBuilder.byClass(DummyPoint).limit(3).build()],
```

Comment:
I have problem with this case. It doesn't collect any shadows and I don't know why.